src/components/views/dialogs/SlidingSyncOptionsDialog.tsx

```diff
@@ -27,47 +27,35 @@ import { useAsyncMemo } from "../../../hooks/useAsyncMemo";
 import { SettingLevel } from "../../../settings/SettingLevel";
 
 /**
- * Check that the server natively supports sliding sync.
+ * Check if the server declares native support for sliding sync via a proxy.
  * @param cli The MatrixClient of the logged in user.
  * @throws if the proxy server is unreachable or not configured to the given homeserver
  */
-async function syncHealthCheck(cli: MatrixClient): Promise<void> {
-    await cli.http.authedRequest(Method.Post, "/sync", undefined, undefined, {
-        localTimeoutMs: 10 * 1000, // 10s
-        prefix: "/_matrix/client/unstable/org.matrix.msc3575",
-    });
-    logger.info("server natively support sliding sync OK");
-}
-
-/**
- * Check that the proxy url is in fact a sliding sync proxy endpoint and it is up.
- * @param endpoint The proxy endpoint url
- * @param hsUrl The homeserver url of the logged in user.
- * @throws if the proxy server is unreachable or not configured to the given homeserver
- */
-async function proxyHealthCheck(endpoint: string, hsUrl?: string): Promise<void> {
+async function nativeSlidingSyncSupportCheck(cli: MatrixClient): Promise<void> {
+    //const proxyUrl = getSlidingSyncWellKnown(cli)?.url;
     const controller = new AbortController();
     const id = window.setTimeout(() => controller.abort(), 10 * 1000); // 10s
-    const res = await fetch(endpoint + "/client/server.json", {
+    const res = await fetch(cli.baseUrl + "/.well-known/matrix/client", {
```

Comment:
The client has a method to fetch the well-known, you should use that instead

Reply:
`cli.getClientWellKnown()` gives me only an `{m.server: Object}`, which is the local `client.json`, not [my client well-known](https://matrix.geraghty.family/.well-known/matrix/client)

Reply:
What is this

> the local `client.json`

you refer to?

`getClientWellKnown` just does a fetch on the same endpoint

<img width="1054" alt="image" src="https://github.com/matrix-org/matrix-react-sdk/assets/2403652/ff6872dd-ab84-425f-9ccd-a9a3fbee45ba">

Reply:
OK, I'm confusing _myself_ now :sweat_smile: Let me start again.

If I do `const wellKnown = cli.getClientWellKnown()` and step, the value of wellKnown is a single `{m.server: Object}`, there are no other keys in there. What am I doing wrong?

Reply:
<img width="1466" alt="image" src="https://github.com/matrix-org/matrix-react-sdk/assets/2403652/c9173967-4d77-4731-a60b-ba11a96ce864">

I have no idea, that's not something that is in the CS spec.

Is your client something ending up with the server well-known?

https://spec.matrix.org/v1.9/server-server-api/#getwell-knownmatrixserver

Check the network tab of your browser devtools to see requests & responses.

Reply:
Thank you for humouring me so far. I'll have a further dig this evening.

Reply:
I've worked out what I was doing wrong, I'm an absolute idiot haha. I've fixed it.